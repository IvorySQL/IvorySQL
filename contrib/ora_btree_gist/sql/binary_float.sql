-- binary_float check

CREATE TABLE binary_floattmp (a binary_float);

\copy binary_floattmp from 'data/binary_float.data'

SET enable_seqscan=on;

SELECT count(*) FROM binary_floattmp WHERE a <  -179.0;

SELECT count(*) FROM binary_floattmp WHERE a <= -179.0;

SELECT count(*) FROM binary_floattmp WHERE a  = -179.0;

SELECT count(*) FROM binary_floattmp WHERE a >= -179.0;

SELECT count(*) FROM binary_floattmp WHERE a >  -179.0;

SELECT count(*) FROM binary_floattmp WHERE a <> -179.0;

SELECT a, a <-> '-179.0' FROM binary_floattmp ORDER BY a <-> '-179.0' LIMIT 3;

CREATE INDEX binary_floatidx ON binary_floattmp USING gist ( a );

SET enable_seqscan=off;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a <  -179.0::binary_float;
SELECT count(*) FROM binary_floattmp WHERE a <  -179.0::binary_float;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a <= -179.0::binary_float;
SELECT count(*) FROM binary_floattmp WHERE a <= -179.0::binary_float;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a  = -179.0::binary_float;
SELECT count(*) FROM binary_floattmp WHERE a  = -179.0::binary_float;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a >= -179.0::binary_float;
SELECT count(*) FROM binary_floattmp WHERE a >= -179.0::binary_float;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a >  -179.0::binary_float;
SELECT count(*) FROM binary_floattmp WHERE a >  -179.0::binary_float;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a <> -179.0::binary_float;
SELECT count(*) FROM binary_floattmp WHERE a <> -179.0::binary_float;

EXPLAIN (COSTS OFF)
SELECT a, a <-> '-179.0' FROM binary_floattmp ORDER BY a <-> '-179.0' LIMIT 3;
SELECT a, a <-> '-179.0' FROM binary_floattmp ORDER BY a <-> '-179.0' LIMIT 3;
