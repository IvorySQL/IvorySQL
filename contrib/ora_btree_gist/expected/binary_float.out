-- binary_float check
CREATE TABLE binary_floattmp (a binary_float);
\copy binary_floattmp from 'data/binary_float.data'
SET enable_seqscan=on;
SELECT count(*) FROM binary_floattmp WHERE a <  -179.0;
 count 
-------
   244
(1 row)

SELECT count(*) FROM binary_floattmp WHERE a <= -179.0;
 count 
-------
   245
(1 row)

SELECT count(*) FROM binary_floattmp WHERE a  = -179.0;
 count 
-------
     1
(1 row)

SELECT count(*) FROM binary_floattmp WHERE a >= -179.0;
 count 
-------
   303
(1 row)

SELECT count(*) FROM binary_floattmp WHERE a >  -179.0;
 count 
-------
   302
(1 row)

SELECT count(*) FROM binary_floattmp WHERE a <> -179.0;
 count 
-------
   546
(1 row)

SELECT a, a <-> '-179.0' FROM binary_floattmp ORDER BY a <-> '-179.0' LIMIT 3;
     a      | ?column?  
------------+-----------
 -179       | 0
 -189.02386 | 10.023865
 -158.17741 | 20.822586
(3 rows)

CREATE INDEX binary_floatidx ON binary_floattmp USING gist ( a );
SET enable_seqscan=off;
EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a <  -179.0::binary_float;
                          QUERY PLAN                          
--------------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_floattmp
         Filter: ((a)::binary_double < '-179'::binary_double)
         ->  Bitmap Index Scan on binary_floatidx
(4 rows)

SELECT count(*) FROM binary_floattmp WHERE a <  -179.0::binary_float;
 count 
-------
   244
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a <= -179.0::binary_float;
                          QUERY PLAN                           
---------------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_floattmp
         Filter: ((a)::binary_double <= '-179'::binary_double)
         ->  Bitmap Index Scan on binary_floatidx
(4 rows)

SELECT count(*) FROM binary_floattmp WHERE a <= -179.0::binary_float;
 count 
-------
   245
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a  = -179.0::binary_float;
                          QUERY PLAN                          
--------------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_floattmp
         Filter: ((a)::binary_double = '-179'::binary_double)
         ->  Bitmap Index Scan on binary_floatidx
(4 rows)

SELECT count(*) FROM binary_floattmp WHERE a  = -179.0::binary_float;
 count 
-------
     1
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a >= -179.0::binary_float;
                          QUERY PLAN                           
---------------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_floattmp
         Filter: ((a)::binary_double >= '-179'::binary_double)
         ->  Bitmap Index Scan on binary_floatidx
(4 rows)

SELECT count(*) FROM binary_floattmp WHERE a >= -179.0::binary_float;
 count 
-------
   303
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a >  -179.0::binary_float;
                          QUERY PLAN                          
--------------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_floattmp
         Filter: ((a)::binary_double > '-179'::binary_double)
         ->  Bitmap Index Scan on binary_floatidx
(4 rows)

SELECT count(*) FROM binary_floattmp WHERE a >  -179.0::binary_float;
 count 
-------
   302
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_floattmp WHERE a <> -179.0::binary_float;
                          QUERY PLAN                           
---------------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_floattmp
         Filter: ((a)::binary_double <> '-179'::binary_double)
         ->  Bitmap Index Scan on binary_floatidx
(4 rows)

SELECT count(*) FROM binary_floattmp WHERE a <> -179.0::binary_float;
 count 
-------
   546
(1 row)

EXPLAIN (COSTS OFF)
SELECT a, a <-> '-179.0' FROM binary_floattmp ORDER BY a <-> '-179.0' LIMIT 3;
                           QUERY PLAN                           
----------------------------------------------------------------
 Limit
   ->  Index Only Scan using binary_floatidx on binary_floattmp
         Order By: (a <-> '-179'::binary_float)
(3 rows)

SELECT a, a <-> '-179.0' FROM binary_floattmp ORDER BY a <-> '-179.0' LIMIT 3;
     a      | ?column?  
------------+-----------
 -179       | 0
 -189.02386 | 10.023865
 -158.17741 | 20.822586
(3 rows)

